src/models/Todo.ts

```diff
@@ -1,17 +1,13 @@
-// TODO: optional properties should come last and shouldn't be mixed with mandatory properties
-export default class Todo {
+export default interface Todo {
   // TODO: how can id be optional?
-  id?: number;
-  text!: string;
+  // ANSWER: You're right, but I did it because I had a situation where I have a Todo without an id
+  // before I send it to the server. Now I found a better way to do it with Omit
+  id: number;
+  text: string;
   // TODO: why is it mandatory? fields like this should have default value.
   // think about the case when you create a new todo, it is not completed by default
-  isCompleted!: boolean;
-  // TODO: objects that return from the server cannot be anything that is not a "primitive" type liek array, string etc.
-  deadlineTime?: Date;
-
-  // TODO: it seems like you don't create an instance using this ctor.
-  // if you don't use it, interface is better because typescript can import it as type only and not value which will help typescript engine with optimiazation process
-  constructor(todo: Todo) {
-    Object.assign(this, todo);
-  }
+  // ANSWER: so you're saying, make it optional and if it doesn't have a value then it is not completed
+  // I know they are both falsey but isn't false preferable than undefined?
```

Comment:
it depends on each case, here logically it's better defining it as optional to save from the client the need to pass false always at creation.

when I say optional you can create default value on ctor to be false and therefore it won't be undefined